ambry-store/src/main/java/com.github.ambry.store/StoreMessageReadSet.java

```diff
@@ -131,13 +127,17 @@ static BlobReadOptions fromBytes(DataInputStream stream, StoreKeyFactory factory
         long size = stream.readLong();
         long expiresAtMs = stream.readLong();
         StoreKey key = factory.getStoreKey(stream);
-        return new BlobReadOptions(log, offset, size, expiresAtMs, key);
+        return new BlobReadOptions(log, offset,
+            new MessageInfo(key, size, expiresAtMs, Account.UNKNOWN_ACCOUNT_ID, Container.UNKNOWN_CONTAINER_ID,
+                Utils.Infinite_Time));
       case VERSION_1:
         offset = Offset.fromBytes(stream);
         size = stream.readLong();
         expiresAtMs = stream.readLong();
         key = factory.getStoreKey(stream);
-        return new BlobReadOptions(log, offset, size, expiresAtMs, key);
+        return new BlobReadOptions(log, offset,
+            new MessageInfo(key, size, expiresAtMs, Account.UNKNOWN_ACCOUNT_ID, Container.UNKNOWN_CONTAINER_ID,
```

Comment:
I guess I'm not too familiar with the HardDelete code, but why does it now require a MessageInfo object?